### ISSUE TITLE
Chat Refactor

### ISSUE THREAD

**Issue by aumu1031** [elementum-games/shinobi-chronicles#336]
* Create `fetchPost` function in `ChatManager` for fetching single post (quote functionality)
* Shift quote logic from server to front-end (react-markdown potentially)
* Move chat post code block (.jsx) into `ChatPost` component
* Make color selection dynamic based on player selections (similar to avatar style)

**Comment by aumu1031** [comment]
Related to #332

**Comment by MiteyDrill** [comment]
Has this been done already?

**Comment by aumu1031** [comment]
> Has this been done already?

Nope, these were just some notes after the old chat was replaced with the React version and given a visual overhaul for New Geisha.
